{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import os\n",
    "import tensorflow as tf\n",
    "import random\n",
    "import re\n",
    "from time import sleep\n",
    "import math\n",
    "import datetime\n",
    "import time\n",
    "import pickle\n",
    "import layers\n",
    "import functions\n",
    "import decoder\n",
    "import encoder\n",
    "from dataset import PubMed_Dataset\n",
    "import data_batcher\n",
    "import models\n",
    "import summarizer\n",
    "import data_utils\n",
    "from sys import stdout\n",
    "from vocab import Vocab_Lookup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "class Binary_Text_Classifier:\n",
    "    def __init__(self, net, thresh=0.5, lr=0.001, mode='train'):\n",
    "        self.thresh = thresh\n",
    "        self.lr = lr\n",
    "        \n",
    "        self._inputs = net.inputs\n",
    "        self._input_lens = net.input_lens\n",
    "        self._dropout_keep_prob = net.dropout_keep_prob\n",
    "        self._logits = net.logits\n",
    "        self._max_len = net.max_len\n",
    "        \n",
    "        self._targets = tf.placeholder(tf.float32, [None, 1])\n",
    "    \n",
    "        self._sigmoid = tf.nn.sigmoid(self._logits)\n",
    "        self._predictions = tf.to_float(self._sigmoid >= self.thresh)\n",
    "        \n",
    "        self._loss = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(labels=self._targets, logits=self._logits))\n",
    "        self._accuracy = tf.reduce_mean(tf.to_float(tf.equal(self._predictions, self._targets)))\n",
    "                \n",
    "        params = tf.trainable_variables()\n",
    "        gradients, _ = tf.clip_by_global_norm(tf.gradients(self._loss, params), 1) \n",
    "        self._gradient_norm = tf.global_norm(gradients)\n",
    "        opt_func = tf.train.AdamOptimizer(learning_rate=self.lr)\n",
    "        self._optimizer = opt_func.apply_gradients(zip(gradients, params)) \n",
    "                \n",
    "    def train_step(self, sess, inputs, input_lens, targets, dropout_keep_prob=1.0):\n",
    "        run_vars = [self._loss, self._accuracy, self._gradient_norm, self._optimizer]\n",
    "        loss, accuracy, grad_norm, _ = sess.run(run_vars, feed_dict={self._inputs : inputs,\n",
    "                                                                     self._input_lens : input_lens,\n",
    "                                                                     self._targets : targets,\n",
    "                                                                     self._dropout_keep_prob : dropout_keep_prob})\n",
    "        return loss, accuracy, grad_norm\n",
    "        \n",
    "    def val_step(self, sess, inputs, input_lens, targets):\n",
    "        run_vars = [self._loss, self._accuracy]\n",
    "        loss, accuracy = sess.run(run_vars, feed_dict={self._inputs : inputs,\n",
    "                                                       self._input_lens : input_lens,\n",
    "                                                       self._targets : targets,\n",
    "                                                       self._dropout_keep_prob : 1.0})\n",
    "        return loss, accuracy\n",
    "        \n",
    "    def deploy(self, sess, inputs, input_lens):\n",
    "        run_vars = self._predictions\n",
    "        predictions = sess.run(run_vars, feed_dict={self._inputs : inputs,\n",
    "                                                    self._input_lens : input_lens,\n",
    "                                                    self._dropout_keep_prob : 1.0})\n",
    "        return predictions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "meta_dir = os.path.join(os.getcwd(), 'PubMed')\n",
    "log_dir = os.path.join(meta_dir, 'logs')\n",
    "weights_dir = os.path.join(meta_dir, 'weights')\n",
    "params_dir = os.path.join(meta_dir, 'params')\n",
    "data_dir = os.path.join(meta_dir, 'data_cache')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#data = pickle.load(open(\"data.pickle\", \"rb\"))\n",
    "vocab_lookup = pickle.load(open(os.path.join(meta_dir, \"vocab_lookup_30000.pickle\"), \"rb\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "train_files = []\n",
    "val_files = []\n",
    "test_files = []\n",
    "for filename in os.listdir(data_dir):\n",
    "    if 'train' in filename:\n",
    "        train_files.append(os.path.join(data_dir, filename))\n",
    "    elif 'val' in filename:\n",
    "        val_files.append(os.path.join(data_dir, filename))\n",
    "    elif 'test' in filename:\n",
    "        test_files.append(os.path.join(data_dir, filename))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def data_partition_loader(partition_files):\n",
    "    i = 0\n",
    "    while True:\n",
    "        partition_file = partition_files[i]\n",
    "        i += 1\n",
    "        yield pickle.load(open(partition_file, 'rb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "train_partition_loader = data_partition_loader(train_files)\n",
    "val_partition_loader = data_partition_loader(val_files)\n",
    "test_partition_loader = data_partition_loader(test_files)\n",
    "\n",
    "train_data = next(train_partition_loader)\n",
    "val_data = next(val_partition_loader)\n",
    "test_data = next(test_partition_loader)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "batch_size = 128\n",
    "train_batcher = data_batcher.Data_Batcher(train_data, batch_size)\n",
    "val_batcher = data_batcher.Data_Batcher(val_data, batch_size)\n",
    "test_batcher = data_batcher.Data_Batcher(test_data, batch_size)\n",
    "deploy_batcher = data_batcher.Data_Batcher(val_data, 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "d_pad_len = 150\n",
    "s_pad_len = 20\n",
    "embd_dim = 100\n",
    "hidden_size = 256\n",
    "n_layers = 4\n",
    "vocab_size = vocab_lookup.num_words\n",
    "dropout_keep_prob = 0.8\n",
    "bidirectional = False\n",
    "\n",
    "display_interval = 100\n",
    "val_interval = 1000\n",
    "deploy_interval = 1000\n",
    "n_iters = 100000\n",
    "\n",
    "lr = 0.001\n",
    "DEVICE = 1\n",
    "USE_CUDA = True\n",
    "DEBUG_MODE = False\n",
    "\n",
    "# w2v = pickle.load(open(\"w2v_CNN-Dailymail_100.pickle\", \"rb\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pretrained_embeddings = None #functions.create_embeddings(vocab_lookup, w2v)    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ai2-leia/anaconda3/envs/tensorflow/lib/python3.6/site-packages/tensorflow/python/ops/gradients_impl.py:93: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.\n",
      "  \"Converting sparse IndexedSlices to a dense Tensor of unknown shape. \"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Trainable Parameters: 4456385\n",
      "<tf.Variable 'embeddings_layer/embeddings:0' shape=(30000, 100) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_0/gru_cell/gates/kernel:0' shape=(356, 512) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_0/gru_cell/gates/bias:0' shape=(512,) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_0/gru_cell/candidate/kernel:0' shape=(356, 256) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_0/gru_cell/candidate/bias:0' shape=(256,) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_1/gru_cell/gates/kernel:0' shape=(512, 512) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_1/gru_cell/gates/bias:0' shape=(512,) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_1/gru_cell/candidate/kernel:0' shape=(512, 256) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_1/gru_cell/candidate/bias:0' shape=(256,) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_2/gru_cell/gates/kernel:0' shape=(512, 512) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_2/gru_cell/gates/bias:0' shape=(512,) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_2/gru_cell/candidate/kernel:0' shape=(512, 256) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_2/gru_cell/candidate/bias:0' shape=(256,) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_3/gru_cell/gates/kernel:0' shape=(512, 512) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_3/gru_cell/gates/bias:0' shape=(512,) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_3/gru_cell/candidate/kernel:0' shape=(512, 256) dtype=float32_ref>\n",
      "<tf.Variable 'encoder/multi_rnn_cell/cell_3/gru_cell/candidate/bias:0' shape=(256,) dtype=float32_ref>\n",
      "<tf.Variable 'dense_output/output_W:0' shape=(256, 1) dtype=float32_ref>\n",
      "<tf.Variable 'dense_output/output_b:0' shape=(1,) dtype=float32_ref>\n"
     ]
    }
   ],
   "source": [
    "device_name = '/gpu:{}'.format(DEVICE) if USE_CUDA else '/cpu:{}'.format(DEVICE)\n",
    "\n",
    "if USE_CUDA:\n",
    "    os.environ[\"CUDA_DEVICE_ORDER\"] = \"PCI_BUS_ID\"\n",
    "    os.environ[\"CUDA_VISIBLE_DEVICES\"] = \"{}\".format(DEVICE)\n",
    "\n",
    "tf.reset_default_graph()\n",
    "with tf.device(device_name):\n",
    "    net = models.RNN_Classifier(1, vocab_size, s_pad_len, embedding_dim=embd_dim, hidden_size=hidden_size, n_layers=n_layers, \n",
    "                                bidirectional=bidirectional, pretrained_embeddings=pretrained_embeddings, trainable_embeddings=True)\n",
    "    model = Binary_Text_Classifier(net, thresh=0.5, lr=lr, mode='train')\n",
    "    functions.count_params(tf.trainable_variables())\n",
    "    for var in tf.trainable_variables(): print(var)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def replace_words(ids, real_len, rate):\n",
    "    mask = [1 if random.random() < rate else 0 for _ in range(real_len-1)] + [0]*(len(ids)-real_len+1)\n",
    "    cond = (np.array(mask) == 1)\n",
    "    random_vals = [random.randint(4, vocab_size) for _ in range(len(ids))]\n",
    "    return np.where(cond, random_vals, ids)\n",
    "\n",
    "def repeat_words(ids, real_len, rate):\n",
    "    mask = [1 if random.random() < rate else 0 for _ in range(real_len-1)] + [0]*(len(ids)-real_len+1)\n",
    "    shift = [ids[0]] + ids[:-1]\n",
    "    cond = (np.array(mask) == 1)\n",
    "    return np.where(cond, shift, ids)\n",
    "\n",
    "def single_repeated_word(ids, real_len, rate):\n",
    "    mask = [1 if random.random() < rate else 0 for _ in range(real_len-1)] + [0]*(len(ids)-real_len+1)\n",
    "    index = random.randint(0, real_len-1)\n",
    "    inclusion = [ids[index] for _ in range(len(ids))]\n",
    "    cond = (np.array(mask) == 1)\n",
    "    return np.where(cond, inclusion, ids)\n",
    "\n",
    "def shuffle_sequence(ids, real_len):\n",
    "    return random.sample(ids[:real_len-1], real_len-1) + ids[real_len-1:]\n",
    "\n",
    "def get_labeled_examples(examples, rate=0.15):\n",
    "    inputs = []\n",
    "    targets = []\n",
    "    for example in examples:\n",
    "        label = random.randint(0, 1)\n",
    "        if label == 0:\n",
    "            aug_type = random.randint(0, 3)\n",
    "            if aug_type == 0:\n",
    "                input_ids = shuffle_sequence(example.target_ids, example.target_len)\n",
    "            elif aug_type == 1:\n",
    "                input_ids = replace_words(example.target_ids, example.target_len, rate)\n",
    "            elif aug_type == 2:\n",
    "                input_ids = repeat_words(example.target_ids, example.target_len, rate)\n",
    "            elif aug_type == 3:\n",
    "                input_ids = single_repeated_word(example.target_ids, example.target_len, rate)\n",
    "        else:\n",
    "            input_ids = example.target_ids\n",
    "        inputs.append(input_ids)\n",
    "        targets.append([label])\n",
    "    return inputs, targets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0,     1] loss: 0.688, accuracy: 0.500, grad_norm: 0.696\n",
      "[0,   100] loss: 0.695, accuracy: 0.547, grad_norm: 0.711\n",
      "[0,   200] loss: 0.664, accuracy: 0.594, grad_norm: 0.863\n",
      "[0,   300] loss: 0.674, accuracy: 0.570, grad_norm: 0.892\n",
      "[0,   400] loss: 0.581, accuracy: 0.680, grad_norm: 0.737\n",
      "[0,   500] loss: 0.652, accuracy: 0.609, grad_norm: 0.979\n",
      "[0,   600] loss: 0.649, accuracy: 0.562, grad_norm: 1.000\n",
      "[0,   700] loss: 0.627, accuracy: 0.625, grad_norm: 0.714\n",
      "[0,   800] loss: 0.653, accuracy: 0.594, grad_norm: 0.939\n",
      "[0,   900] loss: 0.641, accuracy: 0.641, grad_norm: 0.563\n",
      "[0,  1000] loss: 0.663, accuracy: 0.656, grad_norm: 1.000\n",
      "Validation - loss: 0.590, accuracy: 0.663\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_1000\n",
      "\n",
      "INPUT:\n",
      "Six-microns-under acts upstream of Draper in the glial phagocytosis of apoptotic neurons.\n",
      "SHUFFLED:\n",
      "microns UNK apoptotic neurons acts . under the six of upstream glial in phagocytosis of EOS PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "six microns under acts upstream of UNK in the glial phagocytosis of apoptotic neurons . EOS PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "six microns under acts upstream of UNK in in glial phagocytosis phagocytosis apoptotic apoptotic . EOS PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "six microns under acts upstream of UNK microns the glial phagocytosis of apoptotic neurons . EOS PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]\n",
      " [ 1.]]\n",
      "[0,  1100] loss: 0.566, accuracy: 0.742, grad_norm: 0.655\n",
      "[0,  1200] loss: 0.636, accuracy: 0.617, grad_norm: 0.699\n",
      "[0,  1300] loss: 0.559, accuracy: 0.672, grad_norm: 0.746\n",
      "[0,  1400] loss: 0.430, accuracy: 0.805, grad_norm: 1.000\n",
      "[0,  1500] loss: 0.507, accuracy: 0.734, grad_norm: 0.511\n",
      "[0,  1600] loss: 0.506, accuracy: 0.773, grad_norm: 0.511\n",
      "[0,  1700] loss: 0.479, accuracy: 0.719, grad_norm: 0.454\n",
      "[0,  1800] loss: 0.461, accuracy: 0.750, grad_norm: 0.440\n",
      "[0,  1900] loss: 0.422, accuracy: 0.773, grad_norm: 0.459\n",
      "[0,  2000] loss: 0.436, accuracy: 0.797, grad_norm: 1.000\n",
      "Validation - loss: 0.481, accuracy: 0.765\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_2000\n",
      "\n",
      "INPUT:\n",
      "Desirable characteristics of a schistosomiasis vaccine: some implications of a cost-effectiveness analysis.\n",
      "SHUFFLED:\n",
      "some implications a effectiveness schistosomiasis of characteristics desirable . cost a of analysis vaccine EOS PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "desirable characteristics otoacoustic a schistosomiasis vaccine some implications of a cost effectiveness analysis . EOS PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "desirable characteristics of a schistosomiasis schistosomiasis some implications of a cost effectiveness analysis . EOS PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "desirable of of a schistosomiasis vaccine some implications of of cost effectiveness analysis . EOS PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]]\n",
      "[0,  2100] loss: 0.508, accuracy: 0.750, grad_norm: 0.721\n",
      "[0,  2200] loss: 0.454, accuracy: 0.773, grad_norm: 0.364\n",
      "[0,  2300] loss: 0.521, accuracy: 0.711, grad_norm: 0.542\n",
      "[0,  2400] loss: 0.521, accuracy: 0.680, grad_norm: 0.426\n",
      "[0,  2500] loss: 0.498, accuracy: 0.758, grad_norm: 0.528\n",
      "[0,  2600] loss: 0.440, accuracy: 0.789, grad_norm: 0.629\n",
      "[0,  2700] loss: 0.456, accuracy: 0.773, grad_norm: 1.000\n",
      "[0,  2800] loss: 0.450, accuracy: 0.781, grad_norm: 1.000\n",
      "[0,  2900] loss: 0.440, accuracy: 0.766, grad_norm: 0.366\n",
      "[0,  3000] loss: 0.464, accuracy: 0.766, grad_norm: 0.394\n",
      "Validation - loss: 0.440, accuracy: 0.791\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_3000\n",
      "\n",
      "INPUT:\n",
      "Mangiferin attenuates MPTP induced dopaminergic neurodegeneration and improves motor impairment, redox balance and Bcl-2/Bax expression in experimental Parkinson's disease mice.\n",
      "SHUFFLED:\n",
      "expression 2 neurodegeneration improves in and balance induced impairment bcl redox experimental bax UNK mptp attenuates motor and dopaminergic EOS\n",
      "REPLACED:\n",
      "UNK attenuates mptp breadth dopaminergic neurodegeneration and improves motor impairment redox balance and done. 2 bax expression in experimental EOS\n",
      "REPEATED:\n",
      "UNK attenuates mptp induced dopaminergic neurodegeneration and improves motor motor redox redox and bcl 2 bax expression in experimental EOS\n",
      "SINGLE REPEAT:\n",
      "UNK attenuates mptp induced dopaminergic neurodegeneration and improves motor impairment redox balance and bcl 2 bax expression in experimental EOS\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 1.]\n",
      " [ 1.]]\n",
      "[0,  3100] loss: 0.546, accuracy: 0.750, grad_norm: 0.567\n",
      "[0,  3200] loss: 0.495, accuracy: 0.758, grad_norm: 0.360\n",
      "[0,  3300] loss: 0.518, accuracy: 0.773, grad_norm: 0.769\n",
      "[0,  3400] loss: 0.405, accuracy: 0.805, grad_norm: 0.345\n",
      "[0,  3500] loss: 0.493, accuracy: 0.734, grad_norm: 0.627\n",
      "[0,  3600] loss: 0.393, accuracy: 0.812, grad_norm: 0.414\n",
      "[0,  3700] loss: 0.485, accuracy: 0.734, grad_norm: 0.499\n",
      "[0,  3800] loss: 0.407, accuracy: 0.789, grad_norm: 1.000\n",
      "[0,  3900] loss: 0.470, accuracy: 0.805, grad_norm: 0.430\n",
      "[0,  4000] loss: 0.405, accuracy: 0.789, grad_norm: 0.421\n",
      "Validation - loss: 0.432, accuracy: 0.793\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_4000\n",
      "\n",
      "INPUT:\n",
      "Determination of the herbicide desmetryne in organised media by adsorptive stripping voltammetry.\n",
      "SHUFFLED:\n",
      "voltammetry determination stripping organised adsorptive herbicide media of in the UNK by . EOS PAD PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "determination ebpbeta the herbicide exists in organised media by adsorptive stripping voltammetry eggs EOS PAD PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "determination of the herbicide herbicide in organised media media by stripping voltammetry . EOS PAD PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "determination of the herbicide . in organised . . adsorptive stripping voltammetry . EOS PAD PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]]\n",
      "[0,  4100] loss: 0.423, accuracy: 0.797, grad_norm: 0.365\n",
      "[0,  4200] loss: 0.401, accuracy: 0.812, grad_norm: 0.557\n",
      "[0,  4300] loss: 0.410, accuracy: 0.852, grad_norm: 0.308\n",
      "[0,  4400] loss: 0.481, accuracy: 0.742, grad_norm: 0.310\n",
      "[0,  4500] loss: 0.427, accuracy: 0.797, grad_norm: 0.542\n",
      "[0,  4600] loss: 0.346, accuracy: 0.875, grad_norm: 0.321\n",
      "[0,  4700] loss: 0.451, accuracy: 0.812, grad_norm: 0.438\n",
      "[0,  4800] loss: 0.349, accuracy: 0.852, grad_norm: 0.309\n",
      "[0,  4900] loss: 0.354, accuracy: 0.852, grad_norm: 0.281\n",
      "[0,  5000] loss: 0.404, accuracy: 0.836, grad_norm: 0.303\n",
      "Validation - loss: 0.410, accuracy: 0.809\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_5000\n",
      "\n",
      "INPUT:\n",
      "Reliability and validity of the Arabic version of \"dermatitis family impact\" questionnaire in children with atopic dermatitis.\n",
      "SHUFFLED:\n",
      "the with dermatitis of . in family children dermatitis and reliability validity of atopic version arabic impact questionnaire EOS PAD\n",
      "REPLACED:\n",
      "aldehyde and validity of the arabic washout of dermatitis family impact questionnaire in children with atopic dermatitis . EOS PAD\n",
      "REPEATED:\n",
      "reliability and validity of the the version of dermatitis family impact questionnaire in children with with dermatitis . EOS PAD\n",
      "SINGLE REPEAT:\n",
      "reliability and validity of the arabic version of dermatitis family impact questionnaire in children questionnaire atopic dermatitis . EOS PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]\n",
      " [ 1.]]\n",
      "[0,  5100] loss: 0.430, accuracy: 0.773, grad_norm: 0.387\n",
      "[0,  5200] loss: 0.401, accuracy: 0.797, grad_norm: 0.275\n",
      "[0,  5300] loss: 0.374, accuracy: 0.836, grad_norm: 0.296\n",
      "[0,  5400] loss: 0.354, accuracy: 0.883, grad_norm: 0.366\n",
      "[0,  5500] loss: 0.451, accuracy: 0.773, grad_norm: 0.453\n",
      "[0,  5600] loss: 0.418, accuracy: 0.852, grad_norm: 0.415\n",
      "[0,  5700] loss: 0.390, accuracy: 0.820, grad_norm: 0.455\n",
      "[0,  5800] loss: 0.380, accuracy: 0.828, grad_norm: 0.320\n",
      "[0,  5900] loss: 0.288, accuracy: 0.875, grad_norm: 0.292\n",
      "[0,  6000] loss: 0.328, accuracy: 0.867, grad_norm: 0.443\n",
      "Validation - loss: 0.397, accuracy: 0.821\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_6000\n",
      "\n",
      "INPUT:\n",
      "Phosphorylation of mammalian cytochrome c and cytochrome c oxidase in the regulation of cell destiny: respiration, apoptosis, and human disease.\n",
      "SHUFFLED:\n",
      "regulation cytochrome phosphorylation in mammalian the oxidase respiration apoptosis of cell cytochrome c c of and and human UNK EOS\n",
      "REPLACED:\n",
      "phosphorylation of mammalian cytochrome judgements and cytochrome c rubidium in the regulation of footprinting UNK macroscopic apoptosis and human EOS\n",
      "REPEATED:\n",
      "phosphorylation of mammalian cytochrome c and and c oxidase in the regulation of cell UNK respiration apoptosis and human EOS\n",
      "SINGLE REPEAT:\n",
      "phosphorylation of mammalian cytochrome c and cytochrome c oxidase cell the regulation of cell cell respiration apoptosis and human EOS\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]\n",
      " [ 1.]]\n",
      "[0,  6100] loss: 0.465, accuracy: 0.766, grad_norm: 0.338\n",
      "[0,  6200] loss: 0.426, accuracy: 0.766, grad_norm: 0.372\n",
      "[0,  6300] loss: 0.461, accuracy: 0.797, grad_norm: 0.352\n",
      "[0,  6400] loss: 0.463, accuracy: 0.758, grad_norm: 0.385\n",
      "[0,  6500] loss: 0.419, accuracy: 0.805, grad_norm: 0.450\n",
      "[0,  6600] loss: 0.381, accuracy: 0.828, grad_norm: 0.340\n",
      "[0,  6700] loss: 0.337, accuracy: 0.852, grad_norm: 0.370\n",
      "[0,  6800] loss: 0.393, accuracy: 0.797, grad_norm: 0.330\n",
      "[0,  6900] loss: 0.496, accuracy: 0.781, grad_norm: 0.610\n",
      "[0,  7000] loss: 0.364, accuracy: 0.844, grad_norm: 0.246\n",
      "Validation - loss: 0.390, accuracy: 0.823\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_7000\n",
      "\n",
      "INPUT:\n",
      "Disease with abnormal intracranial vascular network complicated with intracerebral haematoma.\n",
      "SHUFFLED:\n",
      "intracerebral intracranial with network with disease complicated vascular abnormal . haematoma EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "hypomethylation with abnormal intracranial vascular vasodilator complicated with intracerebral haematoma . EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "disease with abnormal abnormal vascular vascular complicated with intracerebral haematoma . EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "disease with abnormal intracranial vascular with complicated with intracerebral haematoma with EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 1.]\n",
      " [ 0.]]\n",
      "[0,  7100] loss: 0.404, accuracy: 0.820, grad_norm: 0.254\n",
      "[0,  7200] loss: 0.360, accuracy: 0.852, grad_norm: 0.260\n",
      "[0,  7300] loss: 0.385, accuracy: 0.812, grad_norm: 0.293\n",
      "[0,  7400] loss: 0.376, accuracy: 0.812, grad_norm: 0.262\n",
      "[0,  7500] loss: 0.330, accuracy: 0.844, grad_norm: 0.327\n",
      "[0,  7600] loss: 0.329, accuracy: 0.875, grad_norm: 0.227\n",
      "[0,  7700] loss: 0.433, accuracy: 0.797, grad_norm: 0.305\n",
      "[0,  7800] loss: 0.426, accuracy: 0.820, grad_norm: 0.387\n",
      "[0,  7900] loss: 0.366, accuracy: 0.844, grad_norm: 0.544\n",
      "[0,  8000] loss: 0.327, accuracy: 0.867, grad_norm: 0.232\n",
      "Validation - loss: 0.383, accuracy: 0.826\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_8000\n",
      "\n",
      "INPUT:\n",
      "Effect of reperfusion on the uptake of [3H] uridine in the gerbil brain after prolonged ischaemia.\n",
      "SHUFFLED:\n",
      "uptake of of 3h brain in effect ischaemia after uridine . prolonged the gerbil on reperfusion the EOS PAD PAD\n",
      "REPLACED:\n",
      "effect of reperfusion on the uptake of individually uridine in the gerbil brain after prolonged ischaemia . EOS PAD PAD\n",
      "REPEATED:\n",
      "effect of reperfusion reperfusion the uptake of 3h uridine in the gerbil brain after after ischaemia . EOS PAD PAD\n",
      "SINGLE REPEAT:\n",
      "effect of reperfusion on the uptake of 3h uridine of the gerbil brain after prolonged ischaemia . EOS PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]\n",
      " [ 1.]]\n",
      "[0,  8100] loss: 0.469, accuracy: 0.758, grad_norm: 0.316\n",
      "[0,  8200] loss: 0.370, accuracy: 0.836, grad_norm: 0.256\n",
      "[0,  8300] loss: 0.425, accuracy: 0.797, grad_norm: 0.256\n",
      "[0,  8400] loss: 0.366, accuracy: 0.805, grad_norm: 0.252\n",
      "[0,  8500] loss: 0.333, accuracy: 0.859, grad_norm: 0.320\n",
      "[0,  8600] loss: 0.298, accuracy: 0.898, grad_norm: 0.274\n",
      "[0,  8700] loss: 0.352, accuracy: 0.844, grad_norm: 0.261\n",
      "[0,  8800] loss: 0.450, accuracy: 0.781, grad_norm: 0.283\n",
      "[0,  8900] loss: 0.305, accuracy: 0.883, grad_norm: 0.237\n",
      "[0,  9000] loss: 0.299, accuracy: 0.875, grad_norm: 0.213\n",
      "Validation - loss: 0.381, accuracy: 0.826\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_9000\n",
      "\n",
      "INPUT:\n",
      "EBNA-1, the major nuclear antigen of Epstein-Barr virus, resembles 'RGG' RNA binding proteins.\n",
      "SHUFFLED:\n",
      "virus 1 resembles binding the rna barr of antigen proteins ebna nuclear UNK . epstein major EOS PAD PAD PAD\n",
      "REPLACED:\n",
      "ebna sons the major nuclear antigen of epstein barr virus resembles UNK rna pokeweed proteins . EOS PAD PAD PAD\n",
      "REPEATED:\n",
      "ebna 1 the major nuclear antigen of epstein barr virus resembles UNK rna binding proteins proteins EOS PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "ebna 1 the major nuclear antigen of epstein barr virus resembles UNK rna binding proteins . EOS PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 1.]]\n",
      "[0,  9100] loss: 0.361, accuracy: 0.820, grad_norm: 0.305\n",
      "[0,  9200] loss: 0.357, accuracy: 0.836, grad_norm: 0.273\n",
      "[0,  9300] loss: 0.301, accuracy: 0.891, grad_norm: 0.226\n",
      "[0,  9400] loss: 0.439, accuracy: 0.797, grad_norm: 0.272\n",
      "[0,  9500] loss: 0.409, accuracy: 0.805, grad_norm: 0.337\n",
      "[0,  9600] loss: 0.364, accuracy: 0.852, grad_norm: 0.212\n",
      "[0,  9700] loss: 0.310, accuracy: 0.875, grad_norm: 0.217\n",
      "[0,  9800] loss: 0.332, accuracy: 0.852, grad_norm: 0.189\n",
      "[0,  9900] loss: 0.309, accuracy: 0.852, grad_norm: 0.213\n",
      "[0, 10000] loss: 0.284, accuracy: 0.906, grad_norm: 0.189\n",
      "Validation - loss: 0.369, accuracy: 0.838\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_10000\n",
      "\n",
      "INPUT:\n",
      "Growth changes of the mandibular body with eruption of mandibular third molars: analysis of anatomical morphometry and quantitative bone mineral content by using radiography.\n",
      "SHUFFLED:\n",
      "anatomical of and third quantitative morphometry mandibular changes mandibular of molars eruption of the bone with body analysis growth EOS\n",
      "REPLACED:\n",
      "growth changes aureus the mandibular body with eruption of mandibular third molars analysis of anatomical morphometry and quantitative bone EOS\n",
      "REPEATED:\n",
      "growth changes of the mandibular body with eruption of mandibular third third analysis of anatomical morphometry and quantitative bone EOS\n",
      "SINGLE REPEAT:\n",
      "growth changes of and mandibular and with eruption of and and molars analysis and anatomical morphometry and quantitative bone EOS\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]]\n",
      "[0, 10100] loss: 0.372, accuracy: 0.828, grad_norm: 0.397\n",
      "[0, 10200] loss: 0.385, accuracy: 0.836, grad_norm: 0.242\n",
      "[0, 10300] loss: 0.356, accuracy: 0.820, grad_norm: 0.267\n",
      "[0, 10400] loss: 0.443, accuracy: 0.789, grad_norm: 0.480\n",
      "[0, 10500] loss: 0.352, accuracy: 0.852, grad_norm: 0.253\n",
      "[0, 10600] loss: 0.429, accuracy: 0.797, grad_norm: 0.236\n",
      "[0, 10700] loss: 0.373, accuracy: 0.836, grad_norm: 0.212\n",
      "[0, 10800] loss: 0.324, accuracy: 0.859, grad_norm: 0.193\n",
      "[0, 10900] loss: 0.376, accuracy: 0.852, grad_norm: 0.205\n",
      "[0, 11000] loss: 0.467, accuracy: 0.789, grad_norm: 0.349\n",
      "Validation - loss: 0.354, accuracy: 0.843\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_11000\n",
      "\n",
      "INPUT:\n",
      "Thirty years and counting: finding meaning in the N400 component of the event-related brain potential (ERP).\n",
      "SHUFFLED:\n",
      "in finding component event meaning erp and . related the the years brain thirty n400 counting of potential EOS PAD\n",
      "REPLACED:\n",
      "thirty carcinogenic sores counting finding meaning in the n400 component of conceptus event related brain potential erp . EOS PAD\n",
      "REPEATED:\n",
      "thirty years and counting counting meaning in the n400 n400 of the event related brain potential erp . EOS PAD\n",
      "SINGLE REPEAT:\n",
      "thirty years and years finding meaning years the years years of the event related brain potential erp years EOS PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 11100] loss: 0.341, accuracy: 0.867, grad_norm: 0.229\n",
      "[0, 11200] loss: 0.417, accuracy: 0.789, grad_norm: 0.265\n",
      "[0, 11300] loss: 0.458, accuracy: 0.766, grad_norm: 0.300\n",
      "[0, 11400] loss: 0.278, accuracy: 0.906, grad_norm: 0.220\n",
      "[0, 11500] loss: 0.326, accuracy: 0.859, grad_norm: 0.258\n",
      "[0, 11600] loss: 0.428, accuracy: 0.789, grad_norm: 0.336\n",
      "[0, 11700] loss: 0.407, accuracy: 0.820, grad_norm: 0.252\n",
      "[0, 11800] loss: 0.338, accuracy: 0.844, grad_norm: 0.242\n",
      "[0, 11900] loss: 0.334, accuracy: 0.852, grad_norm: 0.229\n",
      "[0, 12000] loss: 0.362, accuracy: 0.828, grad_norm: 0.243\n",
      "Validation - loss: 0.333, accuracy: 0.859\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_12000\n",
      "\n",
      "INPUT:\n",
      "Dermacentor variabilis: acquired resistance to ticks in BALB/c mice.\n",
      "SHUFFLED:\n",
      "UNK mice balb acquired c ticks . in UNK to resistance EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "UNK UNK acquired resistance candidates ticks in balb c mice . EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "UNK UNK acquired resistance to ticks in balb c mice . EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "ticks UNK ticks ticks to ticks in balb c mice . EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]]\n",
      "[0, 12100] loss: 0.313, accuracy: 0.883, grad_norm: 0.275\n",
      "[0, 12200] loss: 0.345, accuracy: 0.844, grad_norm: 0.241\n",
      "[0, 12300] loss: 0.373, accuracy: 0.836, grad_norm: 0.320\n",
      "[0, 12400] loss: 0.285, accuracy: 0.914, grad_norm: 0.206\n",
      "[0, 12500] loss: 0.291, accuracy: 0.875, grad_norm: 0.201\n",
      "[0, 12600] loss: 0.324, accuracy: 0.852, grad_norm: 0.222\n",
      "[0, 12700] loss: 0.332, accuracy: 0.859, grad_norm: 0.229\n",
      "[0, 12800] loss: 0.378, accuracy: 0.836, grad_norm: 0.204\n",
      "[0, 12900] loss: 0.326, accuracy: 0.859, grad_norm: 0.291\n",
      "[0, 13000] loss: 0.348, accuracy: 0.859, grad_norm: 0.279\n",
      "Validation - loss: 0.343, accuracy: 0.852\n",
      "INPUT:\n",
      "ETB2 receptor subtype stimulation relaxes the iris sphincter muscle.\n",
      "SHUFFLED:\n",
      "subtype the UNK iris stimulation . sphincter UNK muscle receptor EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "UNK receptor subtype stimulation UNK the iris sphincter muscle . EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "UNK receptor subtype stimulation UNK the iris sphincter muscle . EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "UNK receptor subtype muscle UNK muscle iris sphincter muscle . EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 13100] loss: 0.256, accuracy: 0.914, grad_norm: 0.235\n",
      "[0, 13200] loss: 0.295, accuracy: 0.859, grad_norm: 0.267\n",
      "[0, 13300] loss: 0.362, accuracy: 0.859, grad_norm: 0.538\n",
      "[0, 13400] loss: 0.331, accuracy: 0.859, grad_norm: 0.275\n",
      "[0, 13500] loss: 0.315, accuracy: 0.891, grad_norm: 0.207\n",
      "[0, 13600] loss: 0.307, accuracy: 0.844, grad_norm: 0.294\n",
      "[0, 13700] loss: 0.416, accuracy: 0.820, grad_norm: 0.293\n",
      "[0, 13800] loss: 0.387, accuracy: 0.812, grad_norm: 0.210\n",
      "[0, 13900] loss: 0.444, accuracy: 0.805, grad_norm: 0.346\n",
      "[0, 14000] loss: 0.263, accuracy: 0.898, grad_norm: 0.192\n",
      "Validation - loss: 0.321, accuracy: 0.868\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_14000\n",
      "\n",
      "INPUT:\n",
      "Fe-Ce-ZSM-5 a new catalyst of outstanding properties in the selective catalytic reduction of NO with NH(3).\n",
      "SHUFFLED:\n",
      "properties UNK catalytic new fe nh with 5 no ce catalyst of reduction of a selective in outstanding the EOS\n",
      "REPLACED:\n",
      "fe ce legs 5 a new snc of outstanding properties postures the selective catalytic con ht1a no with silence EOS\n",
      "REPEATED:\n",
      "fe ce UNK 5 a new catalyst of of properties in in selective catalytic reduction of no no nh EOS\n",
      "SINGLE REPEAT:\n",
      "fe ce UNK 5 a new catalyst of outstanding properties in the selective catalytic reduction of of with nh EOS\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 14100] loss: 0.368, accuracy: 0.828, grad_norm: 0.230\n",
      "[0, 14200] loss: 0.347, accuracy: 0.859, grad_norm: 0.269\n",
      "[0, 14300] loss: 0.257, accuracy: 0.914, grad_norm: 0.199\n",
      "[0, 14400] loss: 0.413, accuracy: 0.844, grad_norm: 0.298\n",
      "[0, 14500] loss: 0.363, accuracy: 0.859, grad_norm: 0.238\n",
      "[0, 14600] loss: 0.321, accuracy: 0.828, grad_norm: 0.258\n",
      "[0, 14700] loss: 0.330, accuracy: 0.859, grad_norm: 0.313\n",
      "[0, 14800] loss: 0.406, accuracy: 0.781, grad_norm: 0.307\n",
      "[0, 14900] loss: 0.366, accuracy: 0.844, grad_norm: 0.266\n",
      "[0, 15000] loss: 0.309, accuracy: 0.867, grad_norm: 0.261\n",
      "Validation - loss: 0.328, accuracy: 0.863\n",
      "INPUT:\n",
      "Functional and structural relevance of conserved positively charged lysine residues in organic anion transporting polypeptide 1B3.\n",
      "SHUFFLED:\n",
      "relevance in organic residues anion transporting . lysine of charged functional conserved UNK positively and polypeptide structural EOS PAD PAD\n",
      "REPLACED:\n",
      "functional and structural relevance of conserved positively charged lysine dibenzofurans prairie organic anion transporting charcoal UNK . EOS PAD PAD\n",
      "REPEATED:\n",
      "functional and structural relevance of conserved positively positively lysine residues in organic anion transporting transporting UNK . EOS PAD PAD\n",
      "SINGLE REPEAT:\n",
      "functional and structural relevance of transporting transporting transporting lysine residues in organic transporting transporting polypeptide UNK . EOS PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 15100] loss: 0.357, accuracy: 0.836, grad_norm: 0.232\n",
      "[0, 15200] loss: 0.294, accuracy: 0.883, grad_norm: 0.185\n",
      "[0, 15300] loss: 0.357, accuracy: 0.844, grad_norm: 0.230\n",
      "[0, 15400] loss: 0.285, accuracy: 0.875, grad_norm: 0.206\n",
      "[0, 15500] loss: 0.292, accuracy: 0.898, grad_norm: 0.216\n",
      "[0, 15600] loss: 0.256, accuracy: 0.891, grad_norm: 0.266\n",
      "[0, 15700] loss: 0.418, accuracy: 0.820, grad_norm: 0.332\n",
      "[0, 15800] loss: 0.257, accuracy: 0.891, grad_norm: 0.173\n",
      "[0, 15900] loss: 0.369, accuracy: 0.820, grad_norm: 0.287\n",
      "[0, 16000] loss: 0.277, accuracy: 0.914, grad_norm: 0.189\n",
      "Validation - loss: 0.329, accuracy: 0.864\n",
      "INPUT:\n",
      "Primary and secondary structure of a 360-nucleotide isolate of potato spindle tuber viroid.\n",
      "SHUFFLED:\n",
      "structure potato primary a and of isolate nucleotide spindle tuber secondary UNK 360 . of EOS PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "primary and secondary structure of ceftriaxone menstrual nucleotide isolate of potato spindle tuber UNK . EOS PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "primary and secondary structure of a 360 nucleotide isolate isolate potato spindle tuber UNK . EOS PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "primary UNK secondary structure of a 360 UNK UNK of potato UNK UNK UNK . EOS PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 1.]]\n",
      "[0, 16100] loss: 0.267, accuracy: 0.906, grad_norm: 0.204\n",
      "[0, 16200] loss: 0.301, accuracy: 0.891, grad_norm: 0.255\n",
      "[0, 16300] loss: 0.315, accuracy: 0.875, grad_norm: 0.284\n",
      "[0, 16400] loss: 0.335, accuracy: 0.844, grad_norm: 0.294\n",
      "[0, 16500] loss: 0.262, accuracy: 0.914, grad_norm: 0.212\n",
      "[0, 16600] loss: 0.448, accuracy: 0.805, grad_norm: 0.398\n",
      "[0, 16700] loss: 0.270, accuracy: 0.898, grad_norm: 0.170\n",
      "[0, 16800] loss: 0.271, accuracy: 0.883, grad_norm: 0.209\n",
      "[0, 16900] loss: 0.290, accuracy: 0.891, grad_norm: 0.363\n",
      "[0, 17000] loss: 0.297, accuracy: 0.867, grad_norm: 0.201\n",
      "Validation - loss: 0.305, accuracy: 0.875\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_17000\n",
      "\n",
      "INPUT:\n",
      "Daclizumab (Zenapax) inhibits early interleukin-2 receptor signal transduction events.\n",
      "SHUFFLED:\n",
      "early . transduction events interleukin 2 inhibits receptor UNK UNK signal EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "UNK UNK inhibits early coronavirus eighteen receptor signal transduction events . EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "UNK UNK inhibits early interleukin 2 2 receptor transduction events events EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "interleukin UNK inhibits early interleukin interleukin receptor signal transduction events . EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 17100] loss: 0.468, accuracy: 0.812, grad_norm: 0.377\n",
      "[0, 17200] loss: 0.398, accuracy: 0.812, grad_norm: 0.347\n",
      "[0, 17300] loss: 0.238, accuracy: 0.914, grad_norm: 0.239\n",
      "[0, 17400] loss: 0.266, accuracy: 0.891, grad_norm: 0.309\n",
      "[0, 17500] loss: 0.282, accuracy: 0.867, grad_norm: 0.307\n",
      "[0, 17600] loss: 0.306, accuracy: 0.883, grad_norm: 0.208\n",
      "[0, 17700] loss: 0.221, accuracy: 0.930, grad_norm: 0.235\n",
      "[0, 17800] loss: 0.289, accuracy: 0.898, grad_norm: 0.155\n",
      "[0, 17900] loss: 0.260, accuracy: 0.906, grad_norm: 0.209\n",
      "[0, 18000] loss: 0.300, accuracy: 0.891, grad_norm: 0.155\n",
      "Validation - loss: 0.318, accuracy: 0.869\n",
      "INPUT:\n",
      "Influenza and meningococcal disease: lessons for travellers and government from 2 epidemic diseases.\n",
      "SHUFFLED:\n",
      "meningococcal influenza 2 from epidemic lessons . diseases government travellers and for and disease EOS PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "intrapartum and meningococcal disease facet for travellers and government from 2 epidemic diseases acknowledge EOS PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "influenza and meningococcal disease lessons for for and government government 2 epidemic epidemic . EOS PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "influenza and meningococcal disease lessons for travellers and government from 2 epidemic 2 . EOS PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 1.]]\n",
      "[0, 18100] loss: 0.388, accuracy: 0.820, grad_norm: 0.284\n",
      "[0, 18200] loss: 0.289, accuracy: 0.875, grad_norm: 0.205\n",
      "[0, 18300] loss: 0.292, accuracy: 0.883, grad_norm: 0.235\n",
      "[0, 18400] loss: 0.411, accuracy: 0.820, grad_norm: 0.244\n",
      "[0, 18500] loss: 0.289, accuracy: 0.898, grad_norm: 0.179\n",
      "[0, 18600] loss: 0.397, accuracy: 0.828, grad_norm: 0.273\n",
      "[0, 18700] loss: 0.312, accuracy: 0.891, grad_norm: 0.317\n",
      "[0, 18800] loss: 0.388, accuracy: 0.812, grad_norm: 0.306\n",
      "[0, 18900] loss: 0.316, accuracy: 0.859, grad_norm: 0.225\n",
      "[0, 19000] loss: 0.295, accuracy: 0.891, grad_norm: 0.292\n",
      "Validation - loss: 0.321, accuracy: 0.870\n",
      "INPUT:\n",
      "Na+-Ca2+ exchange is affected by membrane potential in cardiac sarcolemmal vesicles.\n",
      "SHUFFLED:\n",
      ". sarcolemmal by is affected vesicles in na potential membrane ca2 exchange cardiac EOS PAD PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "na ca2 exchange is organic by membrane potential mycobacteria cardiac sarcolemmal vesicles . EOS PAD PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "na na exchange is affected by by potential in cardiac sarcolemmal vesicles . EOS PAD PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "na ca2 exchange is affected by membrane exchange in cardiac sarcolemmal vesicles . EOS PAD PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 1.]]\n",
      "[0, 19100] loss: 0.274, accuracy: 0.906, grad_norm: 0.214\n",
      "[0, 19200] loss: 0.369, accuracy: 0.828, grad_norm: 0.247\n",
      "[0, 19300] loss: 0.323, accuracy: 0.852, grad_norm: 0.234\n",
      "[0, 19400] loss: 0.243, accuracy: 0.914, grad_norm: 0.189\n",
      "[0, 19500] loss: 0.265, accuracy: 0.898, grad_norm: 0.192\n",
      "[0, 19600] loss: 0.330, accuracy: 0.875, grad_norm: 0.229\n",
      "[0, 19700] loss: 0.367, accuracy: 0.844, grad_norm: 0.208\n",
      "[0, 19800] loss: 0.393, accuracy: 0.836, grad_norm: 0.219\n",
      "[0, 19900] loss: 0.316, accuracy: 0.867, grad_norm: 0.286\n",
      "[0, 20000] loss: 0.281, accuracy: 0.898, grad_norm: 0.260\n",
      "Validation - loss: 0.312, accuracy: 0.870\n",
      "INPUT:\n",
      "Free fatty acid uptake in humans with CD36 deficiency.\n",
      "SHUFFLED:\n",
      "uptake humans free . cd36 in acid with fatty deficiency EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "blade fatty acid uptake in humans chernobyl cd36 deficiency . EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "free fatty acid uptake in humans with cd36 deficiency . EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "free cd36 acid uptake cd36 humans with cd36 deficiency cd36 EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 1.]\n",
      " [ 0.]]\n",
      "[0, 20100] loss: 0.363, accuracy: 0.859, grad_norm: 0.194\n",
      "[0, 20200] loss: 0.299, accuracy: 0.883, grad_norm: 0.215\n",
      "[0, 20300] loss: 0.325, accuracy: 0.852, grad_norm: 0.330\n",
      "[0, 20400] loss: 0.293, accuracy: 0.875, grad_norm: 0.261\n",
      "[0, 20500] loss: 0.289, accuracy: 0.867, grad_norm: 0.191\n",
      "[0, 20600] loss: 0.320, accuracy: 0.867, grad_norm: 0.247\n",
      "[0, 20700] loss: 0.298, accuracy: 0.867, grad_norm: 0.193\n",
      "[0, 20800] loss: 0.256, accuracy: 0.898, grad_norm: 0.219\n",
      "[0, 20900] loss: 0.339, accuracy: 0.836, grad_norm: 0.252\n",
      "[0, 21000] loss: 0.348, accuracy: 0.852, grad_norm: 0.186\n",
      "Validation - loss: 0.317, accuracy: 0.872\n",
      "INPUT:\n",
      "Experimental pig model of old myocardial infarction with long survival leading to chronic left ventricular dysfunction and remodeling as evaluated by PET.\n",
      "SHUFFLED:\n",
      "as survival of myocardial chronic ventricular experimental left infarction pig old with to leading model dysfunction remodeling and long EOS\n",
      "REPLACED:\n",
      "experimental pig model of isoxazole myocardial infarction with long survival leading to chronic element ventricular dysfunction and remodeling as EOS\n",
      "REPEATED:\n",
      "experimental pig model of old myocardial infarction with long survival survival to chronic left ventricular dysfunction and remodeling as EOS\n",
      "SINGLE REPEAT:\n",
      "experimental pig pig pig old myocardial infarction with long survival leading to chronic left ventricular dysfunction and remodeling as EOS\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 21100] loss: 0.324, accuracy: 0.867, grad_norm: 0.195\n",
      "[0, 21200] loss: 0.336, accuracy: 0.867, grad_norm: 0.280\n",
      "[0, 21300] loss: 0.233, accuracy: 0.922, grad_norm: 0.243\n",
      "[0, 21400] loss: 0.312, accuracy: 0.875, grad_norm: 0.172\n",
      "[0, 21500] loss: 0.294, accuracy: 0.883, grad_norm: 0.194\n",
      "[0, 21600] loss: 0.294, accuracy: 0.875, grad_norm: 0.289\n",
      "[0, 21700] loss: 0.267, accuracy: 0.906, grad_norm: 0.283\n",
      "[0, 21800] loss: 0.328, accuracy: 0.852, grad_norm: 0.178\n",
      "[0, 21900] loss: 0.346, accuracy: 0.867, grad_norm: 0.197\n",
      "[0, 22000] loss: 0.297, accuracy: 0.875, grad_norm: 0.341\n",
      "Validation - loss: 0.300, accuracy: 0.880\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_22000\n",
      "\n",
      "INPUT:\n",
      "Female sterilization: can the woman who will seek reversal be identified prospectively?\n",
      "SHUFFLED:\n",
      "reversal female identified the woman can seek prospectively will be who sterilization EOS PAD PAD PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "female sterilization can the variant annuloplasty will seek reversal be identified prospectively EOS PAD PAD PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "female sterilization sterilization can woman who will seek reversal be identified prospectively EOS PAD PAD PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "female sterilization prospectively the woman who will seek reversal be prospectively prospectively EOS PAD PAD PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 22100] loss: 0.289, accuracy: 0.883, grad_norm: 0.183\n",
      "[0, 22200] loss: 0.314, accuracy: 0.891, grad_norm: 0.360\n",
      "[0, 22300] loss: 0.320, accuracy: 0.852, grad_norm: 0.187\n",
      "[0, 22400] loss: 0.279, accuracy: 0.883, grad_norm: 0.265\n",
      "[0, 22500] loss: 0.399, accuracy: 0.828, grad_norm: 0.297\n",
      "[0, 22600] loss: 0.282, accuracy: 0.898, grad_norm: 0.155\n",
      "[0, 22700] loss: 0.216, accuracy: 0.922, grad_norm: 0.201\n",
      "[0, 22800] loss: 0.287, accuracy: 0.906, grad_norm: 0.196\n",
      "[0, 22900] loss: 0.299, accuracy: 0.867, grad_norm: 0.195\n",
      "[0, 23000] loss: 0.244, accuracy: 0.922, grad_norm: 0.134\n",
      "Validation - loss: 0.299, accuracy: 0.880\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_23000\n",
      "\n",
      "INPUT:\n",
      "Differentiating fall-prone and healthy adults using local dynamic stability.\n",
      "SHUFFLED:\n",
      "fall stability differentiating local prone using healthy dynamic adults and . EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "differentiating fall prone and healthy director ect local las stability southwest EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "differentiating fall prone and healthy adults using local dynamic stability stability EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "differentiating fall prone and healthy adults dynamic local dynamic stability . EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 1.]]\n",
      "[0, 23100] loss: 0.319, accuracy: 0.883, grad_norm: 0.202\n",
      "[0, 23200] loss: 0.349, accuracy: 0.844, grad_norm: 0.244\n",
      "[0, 23300] loss: 0.258, accuracy: 0.898, grad_norm: 0.270\n",
      "[0, 23400] loss: 0.262, accuracy: 0.898, grad_norm: 0.172\n",
      "[0, 23500] loss: 0.384, accuracy: 0.844, grad_norm: 0.188\n",
      "[0, 23600] loss: 0.391, accuracy: 0.820, grad_norm: 0.251\n",
      "[0, 23700] loss: 0.436, accuracy: 0.805, grad_norm: 0.304\n",
      "[0, 23800] loss: 0.336, accuracy: 0.852, grad_norm: 0.247\n",
      "[0, 23900] loss: 0.261, accuracy: 0.891, grad_norm: 0.179\n",
      "[0, 24000] loss: 0.272, accuracy: 0.906, grad_norm: 0.190\n",
      "Validation - loss: 0.287, accuracy: 0.884\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_24000\n",
      "\n",
      "INPUT:\n",
      "Validation of a modified version of the Brief Pain Inventory for painful diabetic peripheral neuropathy.\n",
      "SHUFFLED:\n",
      "brief version . validation the of modified for painful neuropathy of a peripheral pain inventory diabetic EOS PAD PAD PAD\n",
      "REPLACED:\n",
      "validation of a modified version of azido brief pain inventory for painful diabetic peripheral neuropathy . EOS PAD PAD PAD\n",
      "REPEATED:\n",
      "validation validation a modified version version the brief brief inventory for painful diabetic peripheral peripheral . EOS PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "for of a modified version of the brief pain inventory for painful for peripheral neuropathy . EOS PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 24100] loss: 0.196, accuracy: 0.930, grad_norm: 0.175\n",
      "[0, 24200] loss: 0.252, accuracy: 0.898, grad_norm: 0.197\n",
      "[0, 24300] loss: 0.313, accuracy: 0.867, grad_norm: 0.173\n",
      "[0, 24400] loss: 0.240, accuracy: 0.906, grad_norm: 0.217\n",
      "[0, 24500] loss: 0.265, accuracy: 0.906, grad_norm: 0.184\n",
      "[0, 24600] loss: 0.259, accuracy: 0.922, grad_norm: 0.266\n",
      "[0, 24700] loss: 0.270, accuracy: 0.898, grad_norm: 0.157\n",
      "[0, 24800] loss: 0.386, accuracy: 0.805, grad_norm: 0.256\n",
      "[0, 24900] loss: 0.408, accuracy: 0.812, grad_norm: 0.216\n",
      "[0, 25000] loss: 0.293, accuracy: 0.891, grad_norm: 0.196\n",
      "Validation - loss: 0.306, accuracy: 0.875\n",
      "INPUT:\n",
      "Activin A skews macrophage polarization by promoting a proinflammatory phenotype and inhibiting the acquisition of anti-inflammatory macrophage markers.\n",
      "SHUFFLED:\n",
      "macrophage macrophage and the inhibiting inflammatory UNK polarization phenotype anti of activin a acquisition markers proinflammatory a promoting by EOS\n",
      "REPLACED:\n",
      "all. a 2e macrophage polarization by promoting excess proinflammatory phenotype and inhibiting the acquisition of anti inflammatory macrophage neu EOS\n",
      "REPEATED:\n",
      "activin a a macrophage polarization by promoting promoting proinflammatory phenotype and inhibiting the acquisition of anti inflammatory macrophage markers EOS\n",
      "SINGLE REPEAT:\n",
      "activin a UNK polarization polarization by promoting a polarization phenotype and inhibiting polarization acquisition of polarization inflammatory macrophage markers EOS\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 25100] loss: 0.271, accuracy: 0.914, grad_norm: 0.351\n",
      "[0, 25200] loss: 0.305, accuracy: 0.898, grad_norm: 0.168\n",
      "[0, 25300] loss: 0.349, accuracy: 0.844, grad_norm: 0.256\n",
      "[0, 25400] loss: 0.340, accuracy: 0.867, grad_norm: 0.285\n",
      "[0, 25500] loss: 0.303, accuracy: 0.898, grad_norm: 0.263\n",
      "[0, 25600] loss: 0.288, accuracy: 0.852, grad_norm: 0.231\n",
      "[0, 25700] loss: 0.309, accuracy: 0.891, grad_norm: 0.222\n",
      "[0, 25800] loss: 0.281, accuracy: 0.898, grad_norm: 0.195\n",
      "[0, 25900] loss: 0.287, accuracy: 0.883, grad_norm: 0.164\n",
      "[0, 26000] loss: 0.288, accuracy: 0.867, grad_norm: 0.170\n",
      "Validation - loss: 0.304, accuracy: 0.876\n",
      "INPUT:\n",
      "The herpes simplex virus type 1 origin-binding protein interacts specifically with the viral UL8 protein.\n",
      "SHUFFLED:\n",
      "protein type viral UNK protein the herpes simplex origin with . 1 virus the binding specifically interacts EOS PAD PAD\n",
      "REPLACED:\n",
      "radiologists herpes simplex virus type 1 origin binding protein interacts specifically objectives the plasmids UNK protein dbcamp EOS PAD PAD\n",
      "REPEATED:\n",
      "the herpes simplex virus virus 1 origin origin protein interacts specifically with the viral UNK protein . EOS PAD PAD\n",
      "SINGLE REPEAT:\n",
      "the herpes simplex virus type 1 origin binding protein interacts binding with binding viral UNK protein . EOS PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 26100] loss: 0.307, accuracy: 0.852, grad_norm: 0.230\n",
      "[0, 26200] loss: 0.342, accuracy: 0.859, grad_norm: 0.266\n",
      "[0, 26300] loss: 0.259, accuracy: 0.883, grad_norm: 0.166\n",
      "[0, 26400] loss: 0.294, accuracy: 0.875, grad_norm: 0.154\n",
      "[0, 26500] loss: 0.347, accuracy: 0.859, grad_norm: 0.186\n",
      "[0, 26600] loss: 0.256, accuracy: 0.891, grad_norm: 0.158\n",
      "[0, 26700] loss: 0.264, accuracy: 0.898, grad_norm: 0.238\n",
      "[0, 26800] loss: 0.319, accuracy: 0.859, grad_norm: 0.182\n",
      "[0, 26900] loss: 0.329, accuracy: 0.859, grad_norm: 0.184\n",
      "[0, 27000] loss: 0.282, accuracy: 0.891, grad_norm: 0.158\n",
      "Validation - loss: 0.291, accuracy: 0.883\n",
      "INPUT:\n",
      "Dimer structure and conformational variability in the N-terminal region of an archaeal small heat shock protein, StHsp14.0.\n",
      "SHUFFLED:\n",
      "terminal region heat in protein variability an conformational shock UNK and structure n archaeal dimer . the small of EOS\n",
      "REPLACED:\n",
      "fungicide structure and extended variability in the n terminal region phenotype. an archaeal mutational heat shock impart diminished . EOS\n",
      "REPEATED:\n",
      "dimer structure structure conformational conformational in the n terminal region of an an small small heat protein UNK . EOS\n",
      "SINGLE REPEAT:\n",
      "dimer structure and terminal variability in the n terminal region terminal an archaeal small heat shock protein UNK . EOS\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 27100] loss: 0.289, accuracy: 0.883, grad_norm: 0.162\n",
      "[0, 27200] loss: 0.291, accuracy: 0.891, grad_norm: 0.204\n",
      "[0, 27300] loss: 0.243, accuracy: 0.922, grad_norm: 0.205\n",
      "[0, 27400] loss: 0.369, accuracy: 0.859, grad_norm: 0.186\n",
      "[0, 27500] loss: 0.285, accuracy: 0.875, grad_norm: 0.251\n",
      "[0, 27600] loss: 0.260, accuracy: 0.891, grad_norm: 0.266\n",
      "[0, 27700] loss: 0.280, accuracy: 0.883, grad_norm: 0.144\n",
      "[0, 27800] loss: 0.297, accuracy: 0.883, grad_norm: 0.131\n",
      "[0, 27900] loss: 0.345, accuracy: 0.852, grad_norm: 0.184\n",
      "[0, 28000] loss: 0.254, accuracy: 0.914, grad_norm: 0.184\n",
      "Validation - loss: 0.293, accuracy: 0.883\n",
      "INPUT:\n",
      "A PLGA-encapsulated chimeric protein protects against adherence and toxicity of enterotoxigenic Escherichia coli.\n",
      "SHUFFLED:\n",
      ". adherence toxicity chimeric enterotoxigenic escherichia a protein encapsulated coli against protects of and plga EOS PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "a psychoanalysis encapsulated chimeric protein protects against anteroposterior and toxicity of enterotoxigenic escherichia coli . EOS PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "a plga encapsulated chimeric protein protects against adherence and toxicity of enterotoxigenic escherichia escherichia . EOS PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "a plga encapsulated and protein protects against adherence and toxicity and and escherichia coli . EOS PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 28100] loss: 0.288, accuracy: 0.859, grad_norm: 0.203\n",
      "[0, 28200] loss: 0.316, accuracy: 0.875, grad_norm: 0.244\n",
      "[0, 28300] loss: 0.276, accuracy: 0.883, grad_norm: 0.260\n",
      "[0, 28400] loss: 0.370, accuracy: 0.836, grad_norm: 0.287\n",
      "[0, 28500] loss: 0.303, accuracy: 0.875, grad_norm: 0.238\n",
      "[0, 28600] loss: 0.295, accuracy: 0.891, grad_norm: 0.175\n",
      "[0, 28700] loss: 0.205, accuracy: 0.938, grad_norm: 0.264\n",
      "[0, 28800] loss: 0.334, accuracy: 0.836, grad_norm: 0.201\n",
      "[0, 28900] loss: 0.374, accuracy: 0.844, grad_norm: 0.188\n",
      "[0, 29000] loss: 0.320, accuracy: 0.867, grad_norm: 0.249\n",
      "Validation - loss: 0.296, accuracy: 0.881\n",
      "INPUT:\n",
      "Comparison of the electrochemical performance of NiMoO4 nanorods and hierarchical nanospheres for supercapacitor applications.\n",
      "SHUFFLED:\n",
      "of for comparison UNK applications UNK and the hierarchical . of nanospheres performance nanorods electrochemical EOS PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "433 of the electrochemical performance of UNK nanorods and hierarchical nanospheres for malic applications . EOS PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "comparison of the the performance of UNK nanorods and and hierarchical for UNK applications . EOS PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "comparison of the electrochemical performance of UNK nanorods UNK hierarchical nanospheres UNK UNK applications . EOS PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 29100] loss: 0.383, accuracy: 0.844, grad_norm: 0.247\n",
      "[0, 29200] loss: 0.288, accuracy: 0.883, grad_norm: 0.163\n",
      "[0, 29300] loss: 0.249, accuracy: 0.922, grad_norm: 0.210\n",
      "[0, 29400] loss: 0.285, accuracy: 0.883, grad_norm: 0.159\n",
      "[0, 29500] loss: 0.270, accuracy: 0.891, grad_norm: 0.178\n",
      "[0, 29600] loss: 0.390, accuracy: 0.836, grad_norm: 0.284\n",
      "[0, 29700] loss: 0.338, accuracy: 0.844, grad_norm: 0.224\n",
      "[0, 29800] loss: 0.304, accuracy: 0.867, grad_norm: 0.166\n",
      "[0, 29900] loss: 0.362, accuracy: 0.844, grad_norm: 0.237\n",
      "[0, 30000] loss: 0.207, accuracy: 0.938, grad_norm: 0.235\n",
      "Validation - loss: 0.298, accuracy: 0.881\n",
      "INPUT:\n",
      "Endometriosis-associated infertility: evaluation of preoperative use of danazol, gestrinone, and buserelin.\n",
      "SHUFFLED:\n",
      "of preoperative of . evaluation buserelin associated use UNK infertility danazol endometriosis and EOS PAD PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "endometriosis associated infertility limbs. of preoperative use of linolenic UNK and buserelin . EOS PAD PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "endometriosis associated infertility infertility of preoperative use of danazol UNK and buserelin . EOS PAD PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "endometriosis associated infertility evaluation of preoperative use of danazol UNK and buserelin . EOS PAD PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 1.]]\n",
      "[0, 30100] loss: 0.265, accuracy: 0.922, grad_norm: 0.174\n",
      "[0, 30200] loss: 0.275, accuracy: 0.891, grad_norm: 0.185\n",
      "[0, 30300] loss: 0.428, accuracy: 0.812, grad_norm: 0.405\n",
      "[0, 30400] loss: 0.270, accuracy: 0.875, grad_norm: 0.169\n",
      "[0, 30500] loss: 0.220, accuracy: 0.898, grad_norm: 0.171\n",
      "[0, 30600] loss: 0.287, accuracy: 0.891, grad_norm: 0.138\n",
      "[0, 30700] loss: 0.302, accuracy: 0.883, grad_norm: 0.158\n",
      "[0, 30800] loss: 0.325, accuracy: 0.875, grad_norm: 0.247\n",
      "[0, 30900] loss: 0.317, accuracy: 0.909, grad_norm: 0.415\n",
      "[0, 31000] loss: 0.277, accuracy: 0.914, grad_norm: 0.142\n",
      "Validation - loss: 0.286, accuracy: 0.884\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_31000\n",
      "\n",
      "INPUT:\n",
      "Levels of complement anaphylatoxin C5a in pulmonary effluent fluid of infants at risk for chronic lung disease and effects of dexamethasone treatment.\n",
      "SHUFFLED:\n",
      "levels chronic in pulmonary of of UNK effects lung risk and for c5a effluent fluid at disease infants complement EOS\n",
      "REPLACED:\n",
      "levels of complement mitoses igfs in pulmonary effluent fluid of infants at risk for chronic lung disease gbv effects EOS\n",
      "REPEATED:\n",
      "levels of complement complement c5a c5a pulmonary effluent fluid of of at risk for chronic lung disease and effects EOS\n",
      "SINGLE REPEAT:\n",
      "levels of complement UNK c5a in pulmonary and fluid of infants at risk for chronic lung disease and and EOS\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 31100] loss: 0.253, accuracy: 0.914, grad_norm: 0.217\n",
      "[0, 31200] loss: 0.323, accuracy: 0.867, grad_norm: 0.239\n",
      "[0, 31300] loss: 0.288, accuracy: 0.867, grad_norm: 0.160\n",
      "[0, 31400] loss: 0.315, accuracy: 0.875, grad_norm: 0.209\n",
      "[0, 31500] loss: 0.262, accuracy: 0.906, grad_norm: 0.273\n",
      "[0, 31600] loss: 0.290, accuracy: 0.875, grad_norm: 0.226\n",
      "[0, 31700] loss: 0.442, accuracy: 0.820, grad_norm: 0.330\n",
      "[0, 31800] loss: 0.367, accuracy: 0.820, grad_norm: 0.179\n",
      "[0, 31900] loss: 0.212, accuracy: 0.914, grad_norm: 0.167\n",
      "[0, 32000] loss: 0.362, accuracy: 0.859, grad_norm: 0.311\n",
      "Validation - loss: 0.303, accuracy: 0.880\n",
      "INPUT:\n",
      "The increased reaction time of antisaccades. What makes the difference?\n",
      "SHUFFLED:\n",
      "of increased the UNK the makes . time what reaction difference EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "conference increased reaction time interacted bag . what makes the pcc EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "the increased reaction time of of . what makes the difference EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "the UNK reaction time of UNK UNK what makes the difference EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 1.]]\n",
      "[0, 32100] loss: 0.286, accuracy: 0.875, grad_norm: 0.207\n",
      "[0, 32200] loss: 0.296, accuracy: 0.883, grad_norm: 0.229\n",
      "[0, 32300] loss: 0.401, accuracy: 0.844, grad_norm: 0.291\n",
      "[0, 32400] loss: 0.263, accuracy: 0.930, grad_norm: 0.189\n",
      "[0, 32500] loss: 0.293, accuracy: 0.875, grad_norm: 0.168\n",
      "[0, 32600] loss: 0.275, accuracy: 0.898, grad_norm: 0.178\n",
      "[0, 32700] loss: 0.284, accuracy: 0.898, grad_norm: 0.189\n",
      "[0, 32800] loss: 0.253, accuracy: 0.906, grad_norm: 0.206\n",
      "[0, 32900] loss: 0.318, accuracy: 0.883, grad_norm: 0.223\n",
      "[0, 33000] loss: 0.318, accuracy: 0.883, grad_norm: 0.185\n",
      "Validation - loss: 0.292, accuracy: 0.886\n",
      "INPUT:\n",
      "Analogue observational methods in the assessment of social functioning in adults.\n",
      "SHUFFLED:\n",
      "observational methods in analogue functioning . of social adults assessment the in EOS PAD PAD PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "analogue observational methods in the assessment of social functioning 11.4 obesity . EOS PAD PAD PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "analogue observational methods in the assessment of social functioning functioning adults . EOS PAD PAD PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "analogue observational . in the assessment of social functioning in adults . EOS PAD PAD PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 33100] loss: 0.267, accuracy: 0.898, grad_norm: 0.211\n",
      "[0, 33200] loss: 0.274, accuracy: 0.891, grad_norm: 0.204\n",
      "[0, 33300] loss: 0.296, accuracy: 0.898, grad_norm: 0.225\n",
      "[0, 33400] loss: 0.286, accuracy: 0.883, grad_norm: 0.172\n",
      "[0, 33500] loss: 0.310, accuracy: 0.883, grad_norm: 0.169\n",
      "[0, 33600] loss: 0.216, accuracy: 0.906, grad_norm: 0.155\n",
      "[0, 33700] loss: 0.259, accuracy: 0.891, grad_norm: 0.238\n",
      "[0, 33800] loss: 0.303, accuracy: 0.883, grad_norm: 0.215\n",
      "[0, 33900] loss: 0.211, accuracy: 0.922, grad_norm: 0.170\n",
      "[0, 34000] loss: 0.350, accuracy: 0.844, grad_norm: 0.168\n",
      "Validation - loss: 0.288, accuracy: 0.883\n",
      "INPUT:\n",
      "[Methods to optimize the management of Helicobacter pylori infection. The comparative efficacy of the triple and quadruple therapy used as a first line therapy].\n",
      "SHUFFLED:\n",
      "pylori the efficacy the triple and helicobacter to comparative infection quadruple of of therapy the management . methods optimize EOS\n",
      "REPLACED:\n",
      "methods to optimize the management of helicobacter pylori infection . secretions. comparative efficacy antibacterial the velocity. and quadruple therapy EOS\n",
      "REPEATED:\n",
      "methods methods optimize the management of of pylori infection . the comparative comparative of the triple triple quadruple therapy EOS\n",
      "SINGLE REPEAT:\n",
      "comparative to comparative comparative management of comparative pylori infection . the comparative efficacy of the triple and quadruple therapy EOS\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 34100] loss: 0.276, accuracy: 0.883, grad_norm: 0.174\n",
      "[0, 34200] loss: 0.301, accuracy: 0.867, grad_norm: 0.179\n",
      "[0, 34300] loss: 0.224, accuracy: 0.938, grad_norm: 0.131\n",
      "[0, 34400] loss: 0.352, accuracy: 0.852, grad_norm: 0.245\n",
      "[0, 34500] loss: 0.275, accuracy: 0.891, grad_norm: 0.234\n",
      "[0, 34600] loss: 0.260, accuracy: 0.891, grad_norm: 0.142\n",
      "[0, 34700] loss: 0.238, accuracy: 0.922, grad_norm: 0.116\n",
      "[0, 34800] loss: 0.300, accuracy: 0.883, grad_norm: 0.150\n",
      "[0, 34900] loss: 0.282, accuracy: 0.883, grad_norm: 0.223\n",
      "[0, 35000] loss: 0.282, accuracy: 0.883, grad_norm: 0.170\n",
      "Validation - loss: 0.288, accuracy: 0.887\n",
      "INPUT:\n",
      "Validation of a measure for adolescent self-report of attention deficit disorder symptoms.\n",
      "SHUFFLED:\n",
      "adolescent deficit of for validation self measure a disorder report of attention symptoms . EOS PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "validation of a paraplegic germany. adolescent self nymphs of attention deficit disorder symptoms reacting EOS PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "validation validation a measure for adolescent self report of attention deficit disorder symptoms . EOS PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "validation of a measure for adolescent self report of attention deficit adolescent symptoms . EOS PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 35100] loss: 0.261, accuracy: 0.898, grad_norm: 0.195\n",
      "[0, 35200] loss: 0.222, accuracy: 0.938, grad_norm: 0.158\n",
      "[0, 35300] loss: 0.240, accuracy: 0.914, grad_norm: 0.168\n",
      "[0, 35400] loss: 0.266, accuracy: 0.891, grad_norm: 0.129\n",
      "[0, 35500] loss: 0.265, accuracy: 0.898, grad_norm: 0.210\n",
      "[0, 35600] loss: 0.272, accuracy: 0.883, grad_norm: 0.226\n",
      "[0, 35700] loss: 0.330, accuracy: 0.883, grad_norm: 0.234\n",
      "[0, 35800] loss: 0.221, accuracy: 0.922, grad_norm: 0.242\n",
      "[0, 35900] loss: 0.270, accuracy: 0.906, grad_norm: 0.223\n",
      "[0, 36000] loss: 0.296, accuracy: 0.898, grad_norm: 0.181\n",
      "Validation - loss: 0.287, accuracy: 0.887\n",
      "INPUT:\n",
      "Cutaneous vesicostomy: a useful form of temporary diversion in children.\n",
      "SHUFFLED:\n",
      "temporary in . a diversion useful UNK form of cutaneous children EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "cutaneous UNK a tritium form of temporary diversion in mating . EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "cutaneous UNK a useful form of temporary diversion in children . EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "cutaneous UNK a useful form of temporary diversion in diversion diversion EOS PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 1.]\n",
      " [ 0.]]\n",
      "[0, 36100] loss: 0.369, accuracy: 0.852, grad_norm: 0.242\n",
      "[0, 36200] loss: 0.285, accuracy: 0.898, grad_norm: 0.241\n",
      "[0, 36300] loss: 0.375, accuracy: 0.828, grad_norm: 0.210\n",
      "[0, 36400] loss: 0.315, accuracy: 0.875, grad_norm: 0.167\n",
      "[0, 36500] loss: 0.211, accuracy: 0.938, grad_norm: 0.126\n",
      "[0, 36600] loss: 0.293, accuracy: 0.883, grad_norm: 0.180\n",
      "[0, 36700] loss: 0.267, accuracy: 0.898, grad_norm: 0.172\n",
      "[0, 36800] loss: 0.292, accuracy: 0.898, grad_norm: 0.127\n",
      "[0, 36900] loss: 0.255, accuracy: 0.891, grad_norm: 0.208\n",
      "[0, 37000] loss: 0.197, accuracy: 0.938, grad_norm: 0.123\n",
      "Validation - loss: 0.293, accuracy: 0.884\n",
      "INPUT:\n",
      "Choice reaction time after levodopa challenge in parkinsonian patients.\n",
      "SHUFFLED:\n",
      "choice time parkinsonian levodopa after challenge reaction in . patients EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "choice reaction time after levodopa challenge in keratitis patients . EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "choice reaction time after after challenge in parkinsonian patients . EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "choice reaction time after levodopa challenge in reaction patients . EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 37100] loss: 0.267, accuracy: 0.891, grad_norm: 0.182\n",
      "[0, 37200] loss: 0.282, accuracy: 0.859, grad_norm: 0.199\n",
      "[0, 37300] loss: 0.304, accuracy: 0.875, grad_norm: 0.197\n",
      "[0, 37400] loss: 0.287, accuracy: 0.891, grad_norm: 0.181\n",
      "[0, 37500] loss: 0.335, accuracy: 0.867, grad_norm: 0.226\n",
      "[0, 37600] loss: 0.254, accuracy: 0.891, grad_norm: 0.142\n",
      "[0, 37700] loss: 0.250, accuracy: 0.891, grad_norm: 0.177\n",
      "[0, 37800] loss: 0.345, accuracy: 0.852, grad_norm: 0.218\n",
      "[0, 37900] loss: 0.304, accuracy: 0.898, grad_norm: 0.172\n",
      "[0, 38000] loss: 0.338, accuracy: 0.836, grad_norm: 0.221\n",
      "Validation - loss: 0.284, accuracy: 0.887\n",
      "Weights saved in file: /home/ai2-leia/Documents/code/paul/deep_NLP/Summarization/PubMed/weights/RNN_Classifier_checkpoints_20171117_114517/RNN_Classifier_weights_epoch_0_itr_38000\n",
      "\n",
      "INPUT:\n",
      "The curriculum of practice: a conceptual framework for speech-language therapy and audiology practice with a black African first language clientele.\n",
      "SHUFFLED:\n",
      "conceptual practice of curriculum framework with the black a therapy first a practice and speech for UNK african language EOS\n",
      "REPLACED:\n",
      "the curriculum of practice a conceptual framework for speech language therapy and UNK practice with a black african first EOS\n",
      "REPEATED:\n",
      "the curriculum of practice a conceptual framework for speech language therapy and UNK practice with with black black first EOS\n",
      "SINGLE REPEAT:\n",
      "the curriculum and practice a conceptual framework for speech language therapy and UNK practice and a black african first EOS\n",
      "MODEL: [[ 1.]\n",
      " [ 1.]\n",
      " [ 1.]\n",
      " [ 0.]\n",
      " [ 1.]]\n",
      "[0, 38100] loss: 0.237, accuracy: 0.922, grad_norm: 0.214\n",
      "[0, 38200] loss: 0.271, accuracy: 0.898, grad_norm: 0.230\n",
      "[0, 38300] loss: 0.255, accuracy: 0.906, grad_norm: 0.214\n",
      "[0, 38400] loss: 0.273, accuracy: 0.898, grad_norm: 0.181\n",
      "[0, 38500] loss: 0.266, accuracy: 0.906, grad_norm: 0.193\n",
      "[0, 38600] loss: 0.240, accuracy: 0.906, grad_norm: 0.151\n",
      "[0, 38700] loss: 0.281, accuracy: 0.883, grad_norm: 0.209\n",
      "[0, 38800] loss: 0.340, accuracy: 0.852, grad_norm: 0.173\n",
      "[0, 38900] loss: 0.249, accuracy: 0.914, grad_norm: 0.154\n",
      "[0, 39000] loss: 0.203, accuracy: 0.914, grad_norm: 0.201\n",
      "Validation - loss: 0.289, accuracy: 0.885\n",
      "INPUT:\n",
      "Boosting Wnt activity during colorectal cancer progression through selective hypermethylation of Wnt signaling antagonists.\n",
      "SHUFFLED:\n",
      "selective . colorectal cancer during progression wnt antagonists hypermethylation activity signaling of boosting through wnt EOS PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "boosting wnt 294 during colorectal cancer progression through selective hypermethylation of wnt signaling antagonists . EOS PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "boosting wnt activity during colorectal cancer progression progression selective hypermethylation hypermethylation wnt signaling signaling antagonists EOS PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "of wnt of during colorectal cancer progression through selective hypermethylation of of signaling antagonists of EOS PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 1.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 39100] loss: 0.276, accuracy: 0.891, grad_norm: 0.264\n",
      "[0, 39200] loss: 0.239, accuracy: 0.883, grad_norm: 0.372\n",
      "[0, 39300] loss: 0.288, accuracy: 0.891, grad_norm: 0.155\n",
      "[0, 39400] loss: 0.229, accuracy: 0.914, grad_norm: 0.203\n",
      "[0, 39500] loss: 0.293, accuracy: 0.891, grad_norm: 0.191\n",
      "[0, 39600] loss: 0.317, accuracy: 0.891, grad_norm: 0.161\n",
      "[0, 39700] loss: 0.240, accuracy: 0.906, grad_norm: 0.288\n",
      "[0, 39800] loss: 0.235, accuracy: 0.891, grad_norm: 0.198\n",
      "[0, 39900] loss: 0.270, accuracy: 0.898, grad_norm: 0.219\n",
      "[0, 40000] loss: 0.264, accuracy: 0.906, grad_norm: 0.177\n",
      "Validation - loss: 0.290, accuracy: 0.885\n",
      "INPUT:\n",
      "The sociology of Qi Gong: a qualitative study.\n",
      "SHUFFLED:\n",
      ". the qi UNK UNK of study qualitative a EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPLACED:\n",
      "msd UNK wheel qi UNK a vif study . EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "REPEATED:\n",
      "the UNK of qi qi a qualitative qualitative . EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "SINGLE REPEAT:\n",
      "the UNK EOS qi EOS EOS qualitative study EOS EOS PAD PAD PAD PAD PAD PAD PAD PAD PAD PAD\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]]\n",
      "[0, 40100] loss: 0.279, accuracy: 0.883, grad_norm: 0.213\n",
      "[0, 40200] loss: 0.262, accuracy: 0.891, grad_norm: 0.162\n",
      "[0, 40300] loss: 0.242, accuracy: 0.914, grad_norm: 0.226\n",
      "[0, 40400] loss: 0.193, accuracy: 0.945, grad_norm: 0.233\n",
      "[0, 40500] loss: 0.298, accuracy: 0.875, grad_norm: 0.160\n",
      "[0, 40600] loss: 0.244, accuracy: 0.906, grad_norm: 0.262\n",
      "[0, 40700] loss: 0.266, accuracy: 0.914, grad_norm: 0.136\n",
      "[0, 40800] loss: 0.279, accuracy: 0.867, grad_norm: 0.199\n",
      "[0, 40900] loss: 0.282, accuracy: 0.883, grad_norm: 0.173\n",
      "[0, 41000] loss: 0.349, accuracy: 0.828, grad_norm: 0.200\n",
      "Validation - loss: 0.286, accuracy: 0.884\n",
      "INPUT:\n",
      "Association analyses of insulin signaling pathway gene polymorphisms with healthy aging and longevity in Americans of Japanese ancestry.\n",
      "SHUFFLED:\n",
      "signaling pathway americans gene of polymorphisms with in . analyses and aging association insulin japanese healthy ancestry longevity of EOS\n",
      "REPLACED:\n",
      "association dmh of insulin signaling pathway gene polymorphisms with adopted. aging and longevity in americans of japanese ancestry . EOS\n",
      "REPEATED:\n",
      "association analyses of insulin signaling signaling pathway polymorphisms with healthy aging and and in americans of japanese ancestry . EOS\n",
      "SINGLE REPEAT:\n",
      "association with of insulin signaling with gene polymorphisms with with with with longevity in americans of japanese ancestry . EOS\n",
      "MODEL: [[ 1.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]\n",
      " [ 0.]]\n"
     ]
    }
   ],
   "source": [
    "params = {key : value for key, value in net.__dict__.items() if not key.startswith('__') and not key.startswith('_')\n",
    "          and not callable(key) and str(type(value)).find('tensorflow') == -1}\n",
    "model_name = net.__class__.__name__\n",
    "\n",
    "if not DEBUG_MODE:\n",
    "    timestamp = datetime.datetime.fromtimestamp(time.time()).strftime('%Y%m%d_%H%M%S')\n",
    "    log_file = os.path.join(log_dir, '{}_train_log_{}.txt'.format(model_name, timestamp))\n",
    "    log_description = '0.8 dropout, classifier on titles, params: {}\\n'.format(params)\n",
    "    log = open(log_file, 'w')\n",
    "    log.close()\n",
    "    functions.write_to_log(log_description, log_file)\n",
    "    \n",
    "    params_filename = '{}_params_{}.pickle'.format(model_name, timestamp)\n",
    "    with open(os.path.join(params_dir, params_filename), 'wb') as handle:\n",
    "        pickle.dump(params, handle, protocol=pickle.HIGHEST_PROTOCOL)\n",
    "        \n",
    "    checkpoint_dir = os.path.join(weights_dir, '{}_checkpoints_{}'.format(model_name, timestamp))\n",
    "    os.mkdir(checkpoint_dir)\n",
    "\n",
    "epoch = 0\n",
    "with tf.Session(config=tf.ConfigProto(allow_soft_placement=True)) as sess:\n",
    "    sess.run(tf.global_variables_initializer())\n",
    "    \n",
    "    if not DEBUG_MODE:\n",
    "        saver = tf.train.Saver(max_to_keep=100)\n",
    "     \n",
    "    best_val_loss = 10e6\n",
    "    for itr in range(1, n_iters+1):            \n",
    "        examples, ep = train_batcher.next_batch()\n",
    "        if ep == 1:\n",
    "            try:\n",
    "                train_data = next(train_partition_loader)\n",
    "            except:\n",
    "                epoch += 1\n",
    "                train_partition_loader = data_partition_loader(train_files)\n",
    "                train_data = next(train_partition_loader)\n",
    "            train_batcher = data_batcher.Data_Batcher(train_data, batch_size)\n",
    "            examples, ep = train_batcher.next_batch()\n",
    "            \n",
    "        inputs, targets = get_labeled_examples(examples)\n",
    "        input_lens = [example.target_len for example in examples]\n",
    "\n",
    "        train_loss, train_acc, grad_norm = model.train_step(sess, inputs, input_lens, targets, \n",
    "                                                            dropout_keep_prob=dropout_keep_prob)\n",
    "\n",
    "        if itr % display_interval == 0 or itr == 1:     \n",
    "            log_string = ('[%d, %5d] loss: %.3f, accuracy: %.3f, grad_norm: %.3f' \n",
    "                          % (epoch, itr, train_loss, train_acc, grad_norm))\n",
    "\n",
    "            if not DEBUG_MODE:\n",
    "                functions.write_to_log(log_string, log_file)\n",
    "            print(log_string)\n",
    "            \n",
    "        if itr % val_interval == 0:\n",
    "            val_loss, val_acc = 0.0, 0.0 \n",
    "            for i in range(int(len(val_batcher.data)/val_batcher.batch_size)):\n",
    "                examples, ep = val_batcher.next_batch()\n",
    "                if ep == 1:\n",
    "                    try:\n",
    "                        val_data = next(val_partition_loader)\n",
    "                    except:\n",
    "                        val_partition_loader = data_partition_loader(val_files)\n",
    "                        val_data = next(val_partition_loader)\n",
    "                    val_batcher = data_batcher.Data_Batcher(val_data, batch_size)\n",
    "                    examples, ep = val_batcher.next_batch()\n",
    "                inputs, targets = get_labeled_examples(examples)\n",
    "                input_lens = [example.target_len for example in examples]\n",
    "\n",
    "                val_batch_loss, val_batch_acc = model.val_step(sess, inputs, input_lens, targets)\n",
    "                val_loss += ((val_batch_loss - val_loss)/(i+1))\n",
    "                val_acc += ((val_batch_acc - val_acc)/(i+1))\n",
    "                #val_loss.append(val_batch_loss)\n",
    "                #val_acc.append(val_batch_acc)\n",
    "                if (i+1)*val_batcher.batch_size >= 10000:\n",
    "                    break\n",
    "            #val_loss = np.mean(val_loss)\n",
    "            #val_acc = np.mean(val_acc)\n",
    "            log_string = ('Validation - loss: %.3f, accuracy: %.3f' % (val_loss, val_acc))\n",
    "\n",
    "            if not DEBUG_MODE:\n",
    "                functions.write_to_log(log_string, log_file)\n",
    "            print(log_string)\n",
    "            \n",
    "            if not DEBUG_MODE:\n",
    "                if val_loss < best_val_loss:\n",
    "                    best_val_loss = val_loss\n",
    "                    weights_prefix = '{}_weights_epoch_{}_itr_{}'.format(model_name, epoch, itr)\n",
    "                    log_msg = \"Weights saved in file: {}\\n\".format(os.path.join(checkpoint_dir, weights_prefix))\n",
    "                    print(log_msg)\n",
    "                    saver.save(sess, os.path.join(checkpoint_dir, weights_prefix))\n",
    "                    functions.write_to_log(log_msg, log_file)\n",
    "            \n",
    "        if itr % deploy_interval == 0:\n",
    "            examples, _ = deploy_batcher.next_batch()\n",
    "            example = examples[0]\n",
    "            shuffled_input = shuffle_sequence(example.target_ids, example.target_len)\n",
    "            replaced_input = replace_words(example.target_ids, example.target_len, 0.15)\n",
    "            repeated_input = repeat_words(example.target_ids, example.target_len, 0.15)\n",
    "            single_repeated_input = single_repeated_word(example.target_ids, example.target_len, 0.15)\n",
    "            inputs = [example.target_ids, shuffled_input, replaced_input, repeated_input, single_repeated_input]\n",
    "            input_lens = [example.target_len]*5\n",
    "            \n",
    "            replaced_sentence = ' '.join([vocab_lookup.convert_id2word(idx) for idx in replaced_input])\n",
    "            shuffled_sentence = ' '.join([vocab_lookup.convert_id2word(idx) for idx in shuffled_input])\n",
    "            repeated_sentence = ' '.join([vocab_lookup.convert_id2word(idx) for idx in repeated_input])\n",
    "            single_repeated_sentence = ' '.join([vocab_lookup.convert_id2word(idx) for idx in single_repeated_input])\n",
    "            predictions = model.deploy(sess, inputs, input_lens)\n",
    "            \n",
    "            log_string = ('INPUT:\\n{}\\nSHUFFLED:\\n{}\\nREPLACED:\\n{}\\nREPEATED:\\n{}\\nSINGLE REPEAT:\\n{}\\nMODEL: {}'\n",
    "                          .format(example.target_text, shuffled_sentence, replaced_sentence, repeated_sentence, \n",
    "                                  single_repeated_sentence, predictions))\n",
    "            if not DEBUG_MODE:\n",
    "                functions.write_to_log(log_string, log_file)\n",
    "            print(log_string)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:tensorflow]",
   "language": "python",
   "name": "conda-env-tensorflow-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
